Framework/PythonInterface/mantid/geometry/src/Exports/DetectorInfo.cpp

```diff
@@ -0,0 +1,71 @@
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidKernel/Quat.h"
+#include "MantidKernel/V3D.h"
+#include <boost/python/class.hpp>
+#include <boost/python/copy_const_reference.hpp>
+#include <boost/python/return_value_policy.hpp>
+
+using Mantid::Geometry::DetectorInfo;
+using Mantid::Kernel::Quat;
+using Mantid::Kernel::V3D;
+using namespace boost::python;
+
+void export_DetectorInfo() {
+
+  // Function pointers to distinguish between overloaded versions
+  bool (DetectorInfo::*isMonitor)(const size_t) const =
+      &DetectorInfo::isMonitor;
+
+  bool (DetectorInfo::*isMasked)(const size_t) const = &DetectorInfo::isMasked;
+
+  double (DetectorInfo::*twoTheta)(const size_t) const =
+      &DetectorInfo::twoTheta;
+
+  Mantid::Kernel::V3D (DetectorInfo::*position)(const size_t) const =
+      &DetectorInfo::position;
+
+  Mantid::Kernel::Quat (DetectorInfo::*rotation)(const size_t) const =
+      &DetectorInfo::rotation;
+
+  void (DetectorInfo::*setMasked)(const size_t, bool) =
+      &DetectorInfo::setMasked;
+
+  // Export to Python
+  class_<DetectorInfo, boost::noncopyable>("DetectorInfo", no_init)
+      .def("__len__", &DetectorInfo::size, (arg("self")),
+           "Returns the size of the DetectorInfo, i.e., the number of "
+           "detectors in the instrument.")
+
+      .def("size", &DetectorInfo::size, (arg("self")),
+           "Returns the size of the DetectorInfo, i.e., the number of "
+           "detectors in the instrument.")
+
+      .def("isMonitor", isMonitor, (arg("self"), arg("index")),
+           "Returns True if the detector is a monitor.")
+
+      .def("isMasked", isMasked, (arg("self"), arg("index")),
+           "Returns True if the detector is masked.")
+
+      .def("setMasked", setMasked, (arg("self"), arg("index"), arg("masked")),
+           "Set the mask flag of the detector where the detector is identified "
+           "by 'index'.")
+
+      .def("clearMaskFlags", &DetectorInfo::clearMaskFlags, (arg("self")),
+           "Sets all mask flags to false (unmasked).")
+
+      .def("isEquivalent", &DetectorInfo::isEquivalent,
+           (arg("self"), arg("other")), "Returns True if the content of this "
+                                        "detector is equivalent to the content "
+                                        "of the other detector.")
+
+      .def("twoTheta", twoTheta, (arg("self"), arg("index")),
+           "Returns 2 theta (scattering angle w.r.t beam direction).")
```

Comment:
What about the phi angle?

Reply:
I don't think there is a phi angle method in `DetectorInfo` on the C++ side?

Reply:
I remember that there were issues with this because it it was dependant upon how you define it? Is there a way to compute the equivalent value from something else?

Reply:
Own is correct. It is a deliberate choice to not provide it, since it is very much ambiguous. No method `phi()` should be added, if anything it needs a different name:, If I remember correctly, the old implementation on `Detector` simply returns the phi of spherical coordinates and not the azimuth angle of the scattering process (unless the beam is in z direction).